src/jit/objectalloc.cpp

```diff
@@ -166,34 +355,277 @@ GenTree* ObjectAllocator::MorphAllocObjNodeIntoHelperCall(GenTreeAllocObj* alloc
+        //------------------------------------------------------------------------
+
+        GenTree*   tree        = comp->gtNewLclvNode(lclNum, TYP_STRUCT);
+        const bool isVolatile  = false;
+        const bool isCopyBlock = false;
+        tree = comp->gtNewBlkOpNode(tree, comp->gtNewIconNode(0), structSize, isVolatile, isCopyBlock);
+
+        GenTreeStmt* newStmt = comp->gtNewStmt(tree);
+
+        comp->fgInsertStmtBefore(block, stmt, newStmt);
+    }
+
+    //------------------------------------------------------------------------
+    // *  STMT      void
+    // |  /--*  CNS_INT(h) long
+    // \--*  ASG       long
+    //    \--*  FIELD     long   #PseudoField:0x0
+    //       \--*  ADDR      byref
+    //          \--*  LCL_VAR   struct
+    //------------------------------------------------------------------------
+
+    // Create a local representing the object
+    GenTree* tree = comp->gtNewLclvNode(lclNum, TYP_STRUCT);
+
+    // Add a pseudo-field for the method table pointer and initialize it
+    tree = comp->gtNewOperNode(GT_ADDR, TYP_BYREF, tree);
+    tree = comp->gtNewFieldRef(TYP_I_IMPL, FieldSeqStore::FirstElemPseudoField, tree, 0);
+    tree = comp->gtNewAssignNode(tree, allocObj->gtGetOp1());
+
+    GenTreeStmt* newStmt = comp->gtNewStmt(tree);
 
-    return allocObj;
+    comp->fgInsertStmtBefore(block, stmt, newStmt);
+
+    return lclNum;
 }
 
-#ifdef DEBUG
+//------------------------------------------------------------------------
+// BuildConnGraphVisitor: Check if the current tree is a local variable that can point to an object
+//                        and is escaping. Update the connection graph as necessary.
+//
+// Arguments:
+//    pTree - Tree being visited
+//    data  - Walker data
+//
+// Return Value:
+//    Always returns fgWalkResult::WALK_CONTINUE
+
+Compiler::fgWalkResult ObjectAllocator::BuildConnGraphVisitor(GenTree** pTree, Compiler::fgWalkData* data)
+{
+    GenTree* tree = *pTree;
+    assert(tree);
+
+    var_types type = tree->TypeGet();
+    if (tree->OperGet() == GT_LCL_VAR && (type == TYP_REF || type == TYP_BYREF || type == TYP_I_IMPL))
+    {
+        Compiler*                          compiler = data->compiler;
+        GenTree*                           parent   = data->parent;
+        BuildConnGraphVisitorCallbackData* callbackData =
+            reinterpret_cast<BuildConnGraphVisitorCallbackData*>(data->pCallbackData);
+        ObjectAllocator* objectAllocator = callbackData->m_objectAllocator;
+
+        unsigned int lclNum = tree->AsLclVar()->GetLclNum();
+        assert(tree == data->parentStack->Index(0));
+        assert(data->parent == data->parentStack->Index(1));
+
+        if (objectAllocator->CanLclVarEscapeViaParentStack(data->parentStack, lclNum, data->compiler))
+        {
+            if (!objectAllocator->IsLclVarEscaping(lclNum))
+            {
+                JITDUMP("V%02u first escapes (2) via [%06u]\n", lclNum, tree->gtTreeID);
```

Comment:
Probably don't need the `(2)` here. Also we seem to use `dspTreeID` to dump the IDs.

Reply:
Done.